### ISSUE TITLE
Julia 1.11.1 precompilations and freeze with -O3

### ISSUE THREAD

**Issue by ancapdev** [JuliaLang/julia#56207]
Installed julia 1.11.1 with `juliaup`. Launching with no arguments worked as expected. Launching with `-O3` caused some core packages to precompile, I'm guessing the `-O3` binary code is missing from the system image (I can't remember this being an issue on older versions). Upon entering pkg mode, further packages start precompiling, but no progress is made. Full output:

(unable to copy from the terminal while it's updating, so attached as screenshot)

![Image](https://github.com/user-attachments/assets/d0f11600-ded6-4a46-a9f0-4ee7ac41f6d6)

`ctrl+c` doesn't seem to kill this process either, I had to manually kill it. Starting Julia fresh again with `-O3` and it freezes the same way when entering pkg mode.

**Comment by giordano** [comment]
Does this happen with v1.11.0? Is this related to #56054?

**Comment by ancapdev** [comment]
It could be related, but I haven't experienced this issue on v1.11.0 myself.

**Comment by giordano** [comment]
A `git bisect` would be helpful then.

**Comment by ancapdev** [comment]
I've not had time to bisect, but I've found that if I force precompilation of `Pkg` by `using Pkg` first, then everything works fine, including subsequent sessions. It would seem to be the same issue as #56054 potentially then, and I think it may have been masked in my 1.11.0 install by a startup.jl file I had which used `Pkg` and caused the precompilation there. In trying to diagnose an unrelated issue between first installing 1.11.0 and now installing 1.11.1 I had disabled my startup.jl script. I'm guessing my 1.11.0 install was working because the compile cache had already been populated.
